app/controllers/sessions_controller.rb

```diff
@@ -67,6 +68,22 @@ def authenticate
     redirect_to root_path if signed_in?
   end
 
+  def trusted_launch
+    handle_with(SessionsTrustedLaunch,
+                signup_state: signup_state,
+                success: lambda do
+                  if @handler_result.outputs.user
+                    redirect_back
+                  else
+                    redirect_to signup_url
```

Comment:
When we redirect a user to signup, how do we handle the case where we already have an email address for the user?  (what happens on the first start screen which currently requests a role (which we have) and an email address (which we may have)).  Seems like here we'd maybe need to skip `signup#start` or use a variant of it?  

See related comment later that talks about more decisions about where to send different states of users in this `success` block.

Reply:
That case should not happen, because we auto-log them in if there's a matching verified email 

https://github.com/openstax/accounts/pull/539/files#diff-4d6adcaa9d66afacdff074ac1a555244R21

Reply:
Sorry, by "we already have an email address for the user" I meant an email address that doesn't match an existing verified email.  E.g. someone has an account with "bob@gmail.com" but we get "bob@school.edu" in the trusted data.